crates/topos-clock/src/time.rs

```diff
@@ -0,0 +1,99 @@
+use std::{
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+use chrono::{DateTime, Utc};
+use tokio::{
+    spawn,
+    sync::broadcast,
+    time::{interval_at, Instant},
+};
+use tracing::info;
+
+use crate::{Clock, Error, Event};
+
+pub struct TimeClock {
+    genesis: DateTime<Utc>,
+    current_block: Arc<AtomicU64>,
+    epoch_duration: u64,
+    current_epoch: Arc<AtomicU64>,
+}
+
+impl Clock for TimeClock {
+    fn spawn(mut self) -> Result<broadcast::Receiver<Event>, Error> {
+        let (sender, receiver) = broadcast::channel(100);
```

Comment:
Is the capacity set to `100` for some specific reason or how was this value chosen? Should it be a constant perhaps?